pkg/typesystem/typesystem.go

```diff
@@ -778,6 +809,85 @@ func (t *TypeSystem) validateRelationTypeRestrictions() error {
 	return nil
 }
 
+// ensureNoCyclesInTupleToUsersetDefinitions throws an error on the following models because `viewer` is a cycle.
+// type folder
+//
+//	 relations
+//		define parent: [folder] as self
+//		define viewer as viewer from parent
+//
+// and
+//
+// type folder
+//
+//	 relations
+//		define parent as self
+//		define viewer as viewer from parent
```

Comment:
I'm not sure that this example should be a cycle, yeah? In 1.0 model scenario, if you have the following relationship tuples

`folder:1#parent@organization:x`
`organization:x#viewer@jon`

then that model is totally fine.. However, in a 1.0 model scenario, if you have

`folder:1#parent@folder:1`
then that would create an recursive evaluation cycle, which is accounted for with the `ResolutionDepthExceeded` error. But as far as the 1.0 model is concerned I don't see why it's an invalid model.

Reply:
> Is the issue with the 1.0 model merely because there are no other types defined and thus there are no possible folder#parent relationships that could evaluate?

Yes! That's why i have `if typesys.schemaVersion == SchemaVersion1_0 && len(typesys.typeDefinitions) == 1`

Reply:
Gotcha. Thanks for the clarification 👍 . I figured as much, but I wanted to confirm.

Reply:
Yes, sorry, it wasn't super clear. I added a new test that shouldn't throw an error when there's more than one type.